genie-agent/src/main/java/com/netflix/genie/agent/cli/InfoCommand.java

```diff
@@ -68,42 +90,54 @@ public void run() {
                     .append(beanDefinition.isSingleton() ? ", singleton" : "")
                     .toString();
 
-                System.out.println(
-                    String.format(
-                        "  - %s (%s) [%s]",
-                        beanName,
-                        beanClass,
-                        description
+                messageBuilder
+                    .append(
+                        String.format(
+                            "  - %s (%s) [%s]",
+                            beanName,
+                            beanClass == null ? "?" : beanClass,
+                            description
+                        )
                     )
-                );
+                    .append(System.lineSeparator());
             }
         }
 
         if (infoCommandArguments.getIncludeEnvironment()) {
-            System.out.println("Environment variables:");
+            messageBuilder
+                .append("Environment variables:")
+                .append(System.lineSeparator());
             for (Map.Entry<String, String> envEntry : System.getenv().entrySet()) {
-                System.out.println(
-                    String.format(
-                        "  - %s=%s",
-                        envEntry.getKey(),
-                        envEntry.getValue()
+                messageBuilder
+                    .append(
+                        String.format(
+                            "  - %s=%s",
+                            envEntry.getKey(),
+                            envEntry.getValue()
+                        )
                     )
-                );
+                    .append(System.lineSeparator());
             }
         }
 
         if (infoCommandArguments.isIncludeProperties()) {
```

Comment:
You probably also want to Serialize the spring `Environment` here which would give you everything including the systme properties

Reply:
Oh, didn't know about this, will add.